### DISCUSSION TITLE
Zigbee Human Presence Sensor

### DISCUSSION THREAD

**Discussion by Gongman514** [arendst/Tasmota#18120] [Q&A] [answered]
Hello all,
I am playing with a Zigbee Human Presence Sensor that  Blackadder reviewed on 04 Sep 2022. I have connected to a Sonoff Bridge running Tasmota 12.4.0. I am getting MQTT messages. However, I cannot seem to find info on what they mean. I anyone have a magic decoder? Sample of the messages below. 

Thank you,

Bob

![image](https://user-images.githubusercontent.com/10901296/222974494-7e80b6c2-1269-4755-8581-a227503d22ad.png)

11:49:04.426 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0401":1,"Endpoint":1,"LinkQuality":134}}}

11:49:04.475 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0265":0,"Endpoint":1,"LinkQuality":134}}}

11:49:04.526 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0266":0,"Endpoint":1,"LinkQuality":131}}}

11:49:04.626 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0167":1,"Endpoint":1,"LinkQuality":131}}}

11:49:07.942 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0401":1,"Endpoint":1,"LinkQuality":131}}}

11:49:07.993 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0265":0,"Endpoint":1,"LinkQuality":128}}}

11:49:08.042 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0266":0,"Endpoint":1,"LinkQuality":128}}}

11:49:08.091 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0167":1,"Endpoint":1,"LinkQuality":131}}}

11:49:11.458 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0401":1,"Endpoint":1,"LinkQuality":131}}}

11:49:11.508 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0265":0,"Endpoint":1,"LinkQuality":128}}}

11:49:11.557 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0266":0,"Endpoint":1,"LinkQuality":131}}}

11:49:11.605 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","EF00/0167":1,"Endpoint":1,"LinkQuality":131}}}

**Comment by s-hadinger** [comment]
That's unfortunately Tuya proprietary codes. Can you paste the result of `Zbinfo 0xEFB3`. It's highly possible that folks at Zigbee2MQTT have already decoded it.

**Reply by Gongman514** [reply]
@s-hadinger , here is what I got back. 

13:06:04.303 MQT: tele/ZbBridge/SENSOR = {"ZbInfo":{"0xEFB3":{"Device":"0xEFB3","IEEEAddr":"0xA4C13834CCF83F43","ModelId":"TS0601","Manufacturer":"_TZE200_v6ossqfy","Endpoints":[1,242],"Config":["~01.1"],"Reachable":true,"LastSeen":1233,"LastSeenEpoch":1678038331,"LinkQuality":134}}}

**Reply by s-hadinger** [reply]
Thanks, they do have.

```

            switch (dp) {
            case tuya.dataPoints.HPSZInductionState:
                result = {presence: value === 1};
                break;
            case tuya.dataPoints.HPSZPresenceTime:
                result = {duration_of_attendance: value};
                break;
            case tuya.dataPoints.HPSZLeavingTime:
                result = {duration_of_absence: value};
                break;
            case tuya.dataPoints.HPSZLEDState:
                result = {led_state: value};
                break;
```

with

```
    // Human Presence Sensor Zigbee Radiowave Tuya
    HPSZInductionState: 1,
    HPSZPresenceTime: 101,
    HPSZLeavingTime: 102,
    HPSZLEDState: 103,
```

**Reply by s-hadinger** [reply]
Can you try adding the file `hpsz.zb` in the file system with the following content:

```
#Z2Tv1
# Human presence sensor Zigbee
:TS0601,_TZE200_0u3bj3rc
:TS0601,_TZE200_v6ossqfy
:TS0601,_TZE200_mx6u6l4y
EF00/0401,HPSZInductionState            # 1 or 0, 1 if presence is detected
EF00/0265,HPSZPresenceTime              # Shows the presence duration in minutes
EF00/0266,HPSZLeavingTime               # Shows the duration of the absence in minutes
EF00/0167,HPSZLEDState                  # Turns the onboard LED on or off
```

**Reply by s-hadinger** [reply]
You could as well map the first one to `Occupancy` as a synonym:

```
EF00/0401=0406/0000         # map to Occupancy
```

**Reply by Gongman514** [reply]
Thank you for this @s-hadinger. I added the file. Now I get meaningful messages.

15:46:36.413 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","HPSZInductionState":1,"Endpoint":1,"LinkQuality":141}}}
15:46:36.460 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","HPSZPresenceTime":12,"Endpoint":1,"LinkQuality":141}}}
15:46:36.507 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","HPSZLeavingTime":0,"Endpoint":1,"LinkQuality":141}}}
15:46:36.554 MQT: tele/ZbBridge/SENSOR = {"ZbReceived":{"0xEFB3":{"Device":"0xEFB3","HPSZLEDState":1,"Endpoint":1,"LinkQuality":141}}}

Cheers,

Bob

**Comment by s-hadinger** [comment, marked as answer]
I added the file to the Tasmota repo: https://github.com/arendst/Tasmota/blob/development/tasmota/zigbee/Tuya_hpsz.zb
